nestjs/src/courses/mentors/mentors.controller.ts

```diff
@@ -48,11 +49,27 @@ export class MentorsController {
     @Param('mentorId', ParseIntPipe) mentorId: number,
     @Param('courseId', ParseIntPipe) courseId: number,
     @Req() req: CurrentRequest,
-  ) {
+  ): Promise<MentorDashboardDto[]> {
     const hasAccess = await this.mentorsService.canAccessMentor(req.user, mentorId);
     if (!hasAccess) {
       throw new ForbiddenException();
     }
     return await this.mentorsService.getStudentsTasks(mentorId, courseId);
   }
+
+  @Get('/:mentorId/course/:courseId/random-task')
+  @ApiOperation({ operationId: 'getRandomTask' })
+  @ApiOkResponse({ type: InsertResult })
+  @ApiBadRequestResponse()
+  public async getRandomTask(
+    @Param('mentorId', ParseIntPipe) mentorId: number,
+    @Param('courseId', ParseIntPipe) courseId: number,
+    @Req() req: CurrentRequest,
+  ): Promise<InsertResult> {
+    const hasAccess = await this.mentorsService.canAccessMentor(req.user, mentorId);
```

Comment:
seems I missed when this method was added, but looks like it can be replaced with our `RoleGuard` - seems it supports same logic that was added here

Reply:
Fixed, please check. Is it enough to add `RoleGuard` to controller level or should I also add to every method `@RequiredRoles([Role.Admin])`?